main_test.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/bitrise-io/go-utils/fileutil"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/pathutil"
+	"github.com/bitrise-io/go-utils/sliceutil"
+	"github.com/dutchcoders/goftp"
+	"github.com/stretchr/testify/require"
+)
+
+var filesMustBeUploaded = []string{
```

Comment:
in the step description: `Upload a single file or a folder (recursively) to an FTP server.`
Based on the step inputs i think following cases should be covered by the tests:

- source is a file target is a file path
- source is a file target is a dir path
- source is a dir target is a dir path
- source is a content of a dir target is  dir path